worker/firewaller/firewaller.go

```diff
@@ -1439,6 +1440,7 @@ func (fw *Firewaller) startRelation(rel *params.RemoteRelation, role charm.Relat
 	// Start the worker which will watch the remote relation for things like new networks.
 	// We use ReplaceWorker since the relation may have been removed and we are re-adding it.
 	if err := fw.relationWorkerRunner.StartWorker(tag.Id(), func() (worker.Worker, error) {
+		data.catacomb = catacomb.Catacomb{}
```

Comment:
This is the bug "fix".

Reply:
add data.Kill() before the catacomb reset